soc/arm/st_stm32/common/soc_config.c

```diff
@@ -60,13 +60,43 @@ static int st_stm32_common_config(void)
 	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
 #endif /* LL_APB1_GRP1_PERIPH_DBGMCU */
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32MP1X)
-	HAL_EnableDBGSleepMode();
+#endif /* CONFIG_USE_SEGGER_RTT */
+
+
+/* keeping in mind that debugging draws a lot of power we explcitly disable when not needed */
+#if defined(CONFIG_STM32_ENABLE_DEBUG_SLEEP_STOP)
+#define DBGMCU_OP EnableDBGStopMode
 #else
-	LL_DBGMCU_EnableDBGStopMode();
+#define DBGMCU_OP DisableDBGStopMode
+#endif /* CONFIG_STM32_ENABLE_DEBUG_SLEEP_STOP */
+
+/* depending on part function is either in HAL or LL abstraction */
+#define FUNC(x, y) x ## y
+#define HAL_FUNC(op) FUNC(HAL_, op)()
+#define LL_FUNC(op) FUNC(LL_DBGMCU_, op)()
```

Comment:
Sorry but I'll nack on this. 

I appreciate the fact that you tried to optimize the code but this is hardly readable and this prevents grepping, while we essentially need two function calls.
I know the Cube API inconsistency doesn't help here and there may be drivers where you'll find other inventive way to solve similar issues, but to me it is different in this case since:
1- This is quite simple code in this file, we only need to call 2 functions and there is not great issue of readability like we can face in much more complex drivers like ADC for instance
2- To me this is an abuse of macros. Using functions calls (with of course some ifdefs here and there), we can achieve the same in a more readable style.